src/js/xAxisTimeSeries.js

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // Mix-in grade for time-series x axis
+    // should be used alongside the floe.chartAuthoring.axis
+    // and floe.chartAuthoring.xAxis grades
+    // in charts that require a time-based x axis
+
+    fluid.defaults("floe.chartAuthoring.xAxisTimeSeries", {
+        gradeNames: ["floe.chartAuthoring.xAxis"],
+        model: {
+            dataSet: []
+        },
+        axisOptions: {
+            // Configures the time formatting of the axis ticks
+            // in the getXAxisTickFormat function
+            // See https://github.com/mbostock/d3/wiki/Time-Formatting for
+            // explanation of how time formatting works in D3
+            // and https://github.com/mbostock/d3/wiki/Time-Scales#tickFormat
+            // for how it works in the context of a time-based scale
+            XAxisTimeSeriesTickFormats: {
+                milliseconds: "%.L",
+                seconds: ":%S",
+                minute: "%I:%M",
+                hour: "%I %p",
+                day: "%a %d",
+                firstDayOfMonth: "%b %d",
+                month: "%b",
+                year: "%Y"
+            }
+        },
+        invokers: {
+            getXAxisTickFormat: {
+                funcName: "floe.chartAuthoring.xAxisTimeSeries.getXAxisTickFormat",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    floe.chartAuthoring.xAxisTimeSeries.getXAxisTickFormat = function (that) {
+
+        var tickFormats = that.options.axisOptions.XAxisTimeSeriesTickFormats;
+
+        var customTickFormat = d3.time.format.multi([
```

Comment:
The place this code was copy-pasted from should be directly linked (I know it can be reached from the above links) and the small modification that was made to it should be clearly highlighted, along with the reason for it. I seem to remember this being a review comment previously but presumably this file has been regenerated somehow.